src/Symfony/Bridge/Doctrine/ContainerAwareEventManager.php

```diff
@@ -136,9 +141,26 @@ private function initializeListeners($eventName)
     {
         foreach ($this->listeners[$eventName] as $hash => $listener) {
             if (\is_string($listener)) {
-                $this->listeners[$eventName][$hash] = $this->container->get($listener);
+                $this->listeners[$eventName][$hash] = $listener = $this->container->get($listener);
+
+                $this->methods[$listener] = $this->getMethod($listener, $eventName);
             }
         }
         $this->initialized[$eventName] = true;
     }
+
+    /**
+     * @param object $listener
+     * @param string $event
```

Comment:
this `@param string $event` and the `@return string` bellow can be removed: are already type-hinted, so they don't add any value.